src/rx.rs

```diff
@@ -170,6 +177,13 @@ impl<'a> DerefMut for RxPacket<'a> {
 
 impl<'a> Drop for RxPacket<'a> {
     fn drop(&mut self) {
+        // "Preceding reads and writes cannot be moved past subsequent writes."
+        #[cfg(feature = "stm32f7xx-hal")]
+        atomic::fence(Ordering::Release);
+
+        #[cfg(not(feature = "stm32f7xx-hal"))]
```

Comment:
I'm not sure about using the features like this. I expect we'll want to add support for F1 and F2 (very like F4) and H7 (not so like F4) at some point; either this gate should be `cfg(feature="stm32f4xx-hal")` and we add new gates for each HAL, or we should consider just using the non-compiler `fence()` unconditionally.

Or... could we just make the compiler_fence unconditional and the fence conditional? Since adding the extra compiler fence is no overhead if there's already a `fence()`, right? Maybe in `Cargo.toml` we add a new `fence` feature which is enabled by the F7 (and future H7) targets and that gates `fence()` while everyone gets the `compiler_fence()`.

Reply:
I like the second option better, `compiler_fence` to everyone and `fence` through feature.

Reply:
I agree, the second option seems best. If I'd thought of it first it's probably the only one I would have written down :p